border-image-repeat:round output doesn't match other browsers

brentfulgham opened the pull request:
#### 255da9f55bbe24f4aa7944e2bdd5768d4f0d5f01
<pre>
border-image-repeat:round output doesn&apos;t match other browsers
<a href="https://bugs.webkit.org/show_bug.cgi?id=155955">https://bugs.webkit.org/show_bug.cgi?id=155955</a>
&lt;rdar://problem/28213711&gt;

Reviewed by Tim Nguyen.

The RoundTile and SpaceTile didn&apos;t take the scale factors computed in NinePieceImage,
so would arrive at incorrect counts. This caused the logic to scale images when it
should have left them unscaled, and simply tiled (or spaced) the elements.

This change progresses four WPT tests:
(1) border-image-repeat-round.html
(2) border-image-round-and-stretch.html
(3) border-image-slice-percentage-test.html
(4) border-image-space-001.html

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawTiled):

Canonical link: <a href="https://commits.webkit.org/261903@main">https://commits.webkit.org/261903@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c64f1157127e0324f4633946d2399766f92156bc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/113199 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/22332 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1872 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4975 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/121659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/117328 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23702 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/13510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/6209 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118986 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17604 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/100865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/106286 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/99558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/1405 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/46648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14631 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/89/builds/1446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/98895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/15343 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/13510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/20635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/53437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8317 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/17193 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
Can we see the WPT progressions in this patch?

smfr reviewed (commented):
Does this implement this part of the spec?

> "The middle image's width is scaled by the same factor as the top image unless that factor is zero or infinity, in which case the scaling factor of the bottom is substituted, and failing that, the width is not scaled. The height of the middle image is scaled by the same factor as the left image unless that factor is zero or infinity, in which case the scaling factor of the right image is substituted, and failing that, the height is not scaled."

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-repeat-round.html:
```diff
@@ -6,6 +6,7 @@
     <link rel="help" href="http://www.w3.org/TR/css3-background/#border-images">
     <link rel="match" href="reference/border-image-repeat-round-ref.html">
     <meta name="assert" content="diamonds in corners should be red, and other diamonds should be orange, it should be 4 orange diamonds on each side.">
+    <meta name="fuzzy" content="0-92;0-3435">
```
Please use the longhand form for clarity, here and below.

```suggestion
    <meta name="fuzzy" content="maxDifference=0-92;totalPixels=0-3435">
```

nt1m reviewed (approved):
r=me but please export to WPT

Commit:
border-image-repeat:round output doesn't match other browsers
https://bugs.webkit.org/show_bug.cgi?id=155955
<rdar://problem/28213711>

Reviewed by Tim Nguyen.

The RoundTile and SpaceTile didn't take the scale factors computed in NinePieceImage,
so would arrive at incorrect counts. This caused the logic to scale images when it
should have left them unscaled, and simply tiled (or spaced) the elements.

This change progresses four WPT tests:
(1) border-image-repeat-round.html
(2) border-image-round-and-stretch.html
(3) border-image-slice-percentage-test.html
(4) border-image-space-001.html

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawTiled):

Canonical link: https://commits.webkit.org/261903@main
